ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala

```diff
@@ -178,29 +207,24 @@ class IntegrityChecker[LogResult](commitStrategySupport: CommitStrategySupport[L
               + s" submissionEnvelopeSize=${submissionInfo.submissionEnvelope.size()}"
               + s" writeSetSize=${expectedWriteSet.size}"
           )
-          for {
-            _ <- submissionValidator.validateAndCommit(
+          expectedReadServiceFactory.appendBlock(expectedWriteSet)
+          if (!config.indexOnly)
+            submissionValidator.validateAndCommit(
               submissionInfo.submissionEnvelope,
               submissionInfo.correlationId,
               submissionInfo.recordTimeInstant,
               submissionInfo.participantId,
               reader,
               commitStrategy,
-            )
-            actualWriteSet = queryableWriteSet.getAndClearRecordedWriteSet()
-            orderedActualWriteSet = if (config.sortWriteSet) {
-              actualWriteSet.sortBy(_._1.asReadOnlyByteBuffer())
-            } else {
-              actualWriteSet
-            }
-            _ = expectedReadServiceFactory.appendBlock(expectedWriteSet)
-            _ = actualReadServiceFactory.appendBlock(orderedActualWriteSet)
-            _ = if (config.performByteComparison) {
-              compareWriteSets(expectedWriteSet, orderedActualWriteSet)
-            } else {
-              ()
-            }
-          } yield ()
+            ) map { _ =>
+              val actualWriteSet = queryableWriteSet.getAndClearRecordedWriteSet()
+              val orderedActualWriteSet =
+                if (config.sortWriteSet) actualWriteSet.sortBy(_._1.asReadOnlyByteBuffer())
+                else actualWriteSet
```

Comment:
According to the [Scala style guide](https://docs.scala-lang.org/style/control-structures.html#curly-braces):
```suggestion
                if (config.sortWriteSet)
                  actualWriteSet.sortBy(_._1.asReadOnlyByteBuffer())
                else
                  actualWriteSet
```

Reply:
Done.